extensions/INTEL/SPV_INTEL_global_variable_fpga_decorations.asciidoc

```diff
@@ -0,0 +1,202 @@
+= SPV_INTEL_global_variable_fpga_decorations
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+== Name Strings
+
+SPV_INTEL_global_variable_fpga_decorations
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/KhronosGroup/SPIRV-Registry
+
+== Contributors
+
+- Artem Radzikhovskyy, Intel
+- Michael Kinsner, Intel
+- Joe Garvey, Intel
+- Mohammad Fawaz, Intel
+- Gregory Lueck, Intel
+
+== Notice
+
+Copyright (c) 2021-2023 Intel Corporation. All rights reserved.
+
+== Status
+
+* Complete
+
+== Version
+
+[width="40%",cols="25,25"]
+|========================================
+| Last Modified Date | {docdate}
+| Revision      | 1
+|========================================
+
+== Dependencies
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 2.
+
+This extension requires SPIR-V 1.0.
+
+== Overview
+
+This extension adds decorations that can be applied to global (module scope)
+variables. These decorations are intended to help code generation for
+FPGA devices, they can be ignored by all other consumers of this extension.
+
+== Extension Name
+To use this extension within a SPIR-V module, the following *OpExtension* must
+be present in the module:
+
+----
+OpExtension "SPV_INTEL_global_variable_fpga_decorations"
+----
+
+== New Capabilities
+This extension introduces a new capability:
+
+----
+GlobalVariableFPGADecorationsINTEL
+----
+
+== Token Number Assignments
+
+--
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|GlobalVariableFPGADecorationsINTEL | 6146
+|InitModeINTEL | 6147
+|ImplementInRegisterMapINTEL | 6148
+|====
+--
+
+== Modifications to the SPIR-V Specification, Version 1.6
+
+=== Initialization Mode Qualifier
+
+After Section 3.18, add a new section "3.18a Initialization Mode Qualifier" as follows
+
+Defines how the initialization should be triggered.
+
+Used by InitModeINTEL.
+
+--
+[options="header"]
+|====
+2+^| Initialization Mode Qualifier ^| Enabling Capabilities
+// --- ROW BREAK ---
+| 0
+| *InitOnDeviceReprogramINTEL*
+
+Initialization is performed by reprogramming
+ the device. This may require more frequent reprogramming but may reduce
+ area.
+
+| *GlobalVariableFPGADecorationsINTEL*
+
+// --- ROW BREAK ---
+| 1
+| *InitOnDeviceResetINTEL*
+
+Initialization is performed by sending a reset
+ signal to the device. This may increase area but may reduce reprogramming
+ frequency.
+| *GlobalVariableFPGADecorationsINTEL*
+
+|====
+--
+
+=== Decoration
+
+Modify Section 3.20, Decoration, adding these rows to the Decoration table:
+
+--
+[options="header"]
+|====
+2+^| Decoration 2+^| Extra Operands | Enabling Capabilities
```

Comment:
No action required, but the HTML spec will look a little nicer if the columns are adjusted here.  This is what I've been using for Decoration tables for extensions I've written:

```suggestion
[cols="^4,20,2*5,22",options="header"]
|====
2+^.^| Decoration 2+<.^| Extra Operands | Enabling Capabilities
```